plugin/storage/grpc/factory.go

```diff
@@ -208,5 +215,19 @@ func (f *Factory) Close() error {
 	if f.remoteConn != nil {
 		errs = append(errs, f.remoteConn.Close())
 	}
+	if f.noTracingRemoteConn != nil {
+		errs = append(errs, f.noTracingRemoteConn.Close())
+	}
 	return errors.Join(errs...)
 }
+
+func getTelset(logger *zap.Logger, tracerProvider trace.TracerProvider) component.TelemetrySettings {
+	return component.TelemetrySettings{
+		Logger:         logger,
+		TracerProvider: tracerProvider,
+		// TODO needs to be joined with the metricsFactory
```

Comment:
We should book a ticket for v2 board. I'm thinking the internal telset struct should expose both internal metrics factory and otel meter provider, for cases where the latter is needed.

Reply:
@yurishkuro would you be able to elaborate on why we want to expose the meter provider in `telemetry.Setting`. This factory currently just takes in the `metrics.Factory`. Would we want to refactor it to take the whole setting struct?